### ISSUE TITLE
Could not parse version constraint: >dev

### ISSUE THREAD

**Issue by xsduan** [sdispater/poetry#206]
```bash
$ poetry add celery

Using version ^4.2 for celery

Updating dependencies
Resolving dependencies...

[ValueError]
Could not parse version constraint: >dev

add [-D|--dev] [--git GIT] [--path PATH] [-E|--extras EXTRAS] [--optional] [--python PYTHON] [--platform PLATFORM] [--allow-prereleases] [--dry-run] [--] <name> (<name>)...

```

https://github.com/celery/celery/blob/1a941f97ad9f19fe86c241072d21d1e3e6e19a73/requirements/default.txt

```txt
pytz>dev
...
```

`pip install pytz>dev` yields pytz-2018.4.

**Comment by jacebrowning** [comment]
I have also seen this issue (but only when pointing to a private PyPI).

Even if `>dev` is invalid according to [PEP508](https://www.python.org/dev/peps/pep-0508/), which I think it might be, I think `poetry` should just skip over invalid versions.

**Comment by xsduan** [comment]
> I have also seen this issue (but only when pointing to a private PyPI).

Oh, I forgot to mention, I'm using a devpi mirror with private packages. I heard that devpi doesn't have all the metadata that PyPI does when mirroring, maybe there's an issue there?

**Comment by jacebrowning** [comment]
This private PyPI is running 1.0.0 of https://github.com/stevearc/pypicloud.

**Comment by jacebrowning** [comment]
Is the fix for this part of any preview releases?

**Comment by xsduan** [comment]
this seems to have been released in the 0.11.0a4 preview.

it seems to work now!

**Comment by jacebrowning** [comment]
👍 It's working for me.
